### ISSUE TITLE
sub protocols?

### ISSUE THREAD

**Issue by plicease** [plicease/AnyEvent-WebSocket-Client#27]
This came in from an email from a user.

```
If you use JavaScript (Firefox browser) specified subprotocol
is very simple:
var mySocket = new WebSocket('wss://www.example.com', 'my-subprotocol');
or:
var mySocket = new WebSocket('wss://www.example.com', ['my-subprotocol',my-subprotocol2']);

Whe I use Protocol::WebSocket::Client, I must use this "hack":
my $client = Protocol::WebSocket::Client->new(
   url => 'wss://www.example.com',
   version => '13'
);
$client->{hs}->req->fields->{'sec-websocket-protocol'
}='my-subprotocol';

When I use AnyEvent::WebSocket::Client, I don't know how specify
subprotocol.

```

**Comment by plicease** [comment]
I think this is doable with the public interface.

**Comment by plicease** [comment]
I believe this capability is available in the dev release 0.35_01, now on its way to CPAN.  If it still looks like a good idea in a couple of days I will release a production version.
